File: app/src/main/java/com/example/doan/Models/DashboardSummary.java
package com.example.doan.Models;

import com.google.gson.annotations.SerializedName;
import java.io.Serializable;
import java.util.List;

public class DashboardSummary implements Serializable {

    @SerializedName("totalRevenue")
    private double totalRevenue;

    @SerializedName("totalOrders")
    private long totalOrders;

    @SerializedName("pendingOrders")
    private long pendingOrders;

    @SerializedName("completedOrders")
    private long completedOrders;

    @SerializedName("canceledOrders")
    private long canceledOrders;

    @SerializedName("topSellingDrinks")
    private List<TopSellingDrink> topSellingDrinks;

    public DashboardSummary() {}

    // Getters and Setters
    public double getTotalRevenue() {
        return totalRevenue;
    }

    public void setTotalRevenue(double totalRevenue) {
        this.totalRevenue = totalRevenue;
    }

    public long getTotalOrders() {
        return totalOrders;
    }

    public void setTotalOrders(long totalOrders) {
        this.totalOrders = totalOrders;
    }

    public long getPendingOrders() {
        return pendingOrders;
    }

    public void setPendingOrders(long pendingOrders) {
        this.pendingOrders = pendingOrders;
    }

    public long getCompletedOrders() {
        return completedOrders;
    }

    public void setCompletedOrders(long completedOrders) {
        this.completedOrders = completedOrders;
    }

    public long getCanceledOrders() {
        return canceledOrders;
    }

    public void setCanceledOrders(long canceledOrders) {
        this.canceledOrders = canceledOrders;
    }

    public List<TopSellingDrink> getTopSellingDrinks() {
        return topSellingDrinks;
    }

    public void setTopSellingDrinks(List<TopSellingDrink> topSellingDrinks) {
        this.topSellingDrinks = topSellingDrinks;
    }

    public static class TopSellingDrink implements Serializable {
        @SerializedName("drinkName")
        private String drinkName;

        @SerializedName("totalSold")
        private long totalSold;

        @SerializedName("revenue")
        private double revenue;

        public String getDrinkName() {
            return drinkName;
        }

        public long getTotalSold() {
            return totalSold;
        }

        public double getRevenue() {
            return revenue;
        }
    }
}

File: app/src/main/java/com/example/doan/Models/DashboardSummary.kt
package com.example.doan.Models

import com.google.gson.annotations.SerializedName
import java.io.Serializable

import java.math.BigDecimal

data class DashboardSummary(
    @SerializedName("totalRevenue")
    var totalRevenue: BigDecimal = BigDecimal.ZERO,
    
    @SerializedName("totalOrders")
    var totalOrders: Long = 0,
    
    @SerializedName("pendingOrders")
    var pendingOrders: Long = 0,
    
    @SerializedName("completedOrders")
    var completedOrders: Long = 0,
    
    @SerializedName("canceledOrders")
    var canceledOrders: Long = 0,
    
    @SerializedName("topSellingDrinks")
    var topSellingDrinks: List<TopSellingDrink>? = null,
    
    @SerializedName("topRatedDrinks")
    var topRatedDrinks: List<TopRatedDrink>? = null,
    
    // Thông tin stores được quản lý (cho Manager)
    @SerializedName("managedStores")
    var managedStores: List<ManagedStoreInfo>? = null,
    
    @SerializedName("isAdmin")
    var isAdmin: Boolean? = false
) : Serializable {
    
    data class TopSellingDrink(
        @SerializedName("drinkName")
        val drinkName: String? = null,
        
        @SerializedName("totalSold")
        val totalSold: Long = 0,
        
        @SerializedName("revenue")
        val revenue: BigDecimal = BigDecimal.ZERO
    ) : Serializable
    
    data class TopRatedDrink(
        @SerializedName("drinkId")
        val drinkId: Long = 0,
        
        @SerializedName("drinkName")
        val drinkName: String? = null,
        
        @SerializedName("drinkImage")
        val drinkImage: String? = null,
        
        @SerializedName("averageRating")
        val averageRating: Double = 0.0,
        
        @SerializedName("totalReviews")
        val totalReviews: Long = 0
    ) : Serializable
    
    data class ManagedStoreInfo(
        @SerializedName("id")
        val id: Long = 0,
        
        @SerializedName("storeName")
        val storeName: String? = null,
        
        @SerializedName("address")
        val address: String? = null
    ) : Serializable
}
